### PR TITLE
Remove deprecated variable _PS_PRICE_COMPUTE_PRECISION_

### PR DESCRIPTION
**Pull request by idnovate**
Deprecated `_PS_PRICE_COMPUTE_PRECISION_` instead of `Context::getContext()->getComputingPrecision()` was used. This cause an error when value from the 2 variables didn't match.

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | When the value from `_PS_PRICE_COMPUTE_PRECISION_` and `Context::getContext()->getComputingPrecision()` was different, the order was created as Payment Error
| Type?             | bug fix
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27630)

### CODE REVIEW

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
Hi and thanks for your contribution.
Could you please remove the define in config.inc.php as the constant has been tagged as deprecated in 1.7.7.
If you feel comfortable, you can also do it for `_PS_PRICE_DISPLAY_PRECISION_` which was also deprecated in the same version

**Review by matthieu-rolland — COMMENTED** [review]
thanks @idnovate , this is indeed deprecated and should be removed

@idnovate do you think we should have a reproducible step by step use case for the QA team, so that they can test it?

Since the PR is described as a bug fix...

**Comment by idnovate** [comment]
> thanks @idnovate , this is indeed deprecated and should be removed
> 
> @idnovate do you think we should have a reproducible step by step use case for the QA team, so that they can test it?
> 
> Since the PR is described as a bug fix...

Ok

**Comment by idnovate** [comment]
> Hi and thanks for your contribution. Could you please remove the define in config.inc.php as the constant has been tagged as deprecated in 1.7.7. If you feel comfortable, you can also do it for `_PS_PRICE_DISPLAY_PRECISION_` which was also deprecated in the same version

But this could be a BC break if modules or custom developments use this variable.

**Comment by idnovate** [comment]
> thanks @idnovate , this is indeed deprecated and should be removed
> 
> @idnovate do you think we should have a reproducible step by step use case for the QA team, so that they can test it?
> 
> Since the PR is described as a bug fix...

How to reproduce it and test it?

1. Set `0` decimals for a currency:

image


2. Set `Order status` as logable:

image


3. Place an order with a product **above 999€**:

image


4. Order is created with `Payment error`:

image


> @idnovate Could you create a dedicated issue for validating it by the @PrestaShop/qa-functional team ?

https://github.com/PrestaShop/PrestaShop/issues/27902

**Comment by Progi1984** [comment]
@idnovate Could you create a dedicated issue for validating it by the @PrestaShop/qa-functional team ?

File: classes/PaymentModule.php [hunk 1]
```diff
@@ -1157,7 +1157,7 @@ protected function createOrderCartRules(
                 );
             }
             $remainingValue = $cartRuleReductionAmountConverted - $values[$cartRule->reduction_tax ? 'tax_incl' : 'tax_excl'];
-            $remainingValue = Tools::ps_round($remainingValue, _PS_PRICE_COMPUTE_PRECISION_);
+            $remainingValue = Tools::ps_round($remainingValue, Context::getContext()->getComputingPrecision());
```
Comment (kpodemski) [inline classes/PaymentModule.php:1157]:
```suggestion
            $remainingValue = Tools::ps_round($remainingValue, $computingPrecision);
```

**Review by kpodemski — CHANGES_REQUESTED** [review]
In the `createOrderCartRules` you have this variable already available.

In the first method, you could create if it's in a few places.

**Comment by kpodemski** [comment]
Thanks @idnovate 

I added a little feedback.

Also, this is a duplication of:
https://github.com/PrestaShop/PrestaShop/pull/27436

But I see that you changed the variable in the remaining places, good 👍

**Comment by matks** [comment]
Friendly reminder @idnovate 😉 also this PR requires a git rebase

**Comment by Progi1984** [comment]
@idnovate Could you rebase your PR and remove your merge commits, please ?

**Comment by idnovate** [comment]
> @idnovate Could you rebase your PR and remove your merge commits, please ?

Is it OK now? :confused:

**Comment by Progi1984** [comment]
@idnovate No you have always merge commits. Check here : https://github.com/PrestaShop/PrestaShop/pull/27630/commits.

**Comment by idnovate** [comment]
https://github.com/PrestaShop/PrestaShop/pull/28388

**Pull request closed without merging**